src/frontend/js/components/AddressesManagement/index.tsx

```diff
@@ -0,0 +1,309 @@
+   * update `selectedAddress` state with the address provided
+   * then close the address management form
+   *
+   * @param {Joanie.Address} address
+   */
+  const handleSelect = (address: Joanie.Address) => {
+    setError(undefined);
+    selectAddress(address);
+    handleClose();
+  };
+
+  /**
+   * Ask user to confirm its intention
+   * then make the request to delete the provided address
+   *
+   * @param {Joanie.Address} address
+   */
+  const handleDelete = (address: Joanie.Address) => {
+    setError(undefined);
+    // eslint-disable-next-line no-alert, no-restricted-globals
+    const sure = confirm(
+      intl.formatMessage(messages.deletionConfirmation, { title: address.title }),
+    );
+    if (!address.is_main && sure) {
+      addresses.methods.delete(address.id, {
+        onError: () => setError(intl.formatMessage(messages.actionDeletion)),
+      });
+    }
+  };
+
+  /**
+   * Create a new address according to form values
+   * then update `selectedAddress` state with this new one.
+   * If `save` checkbox input is checked, the address is persisted
+   * otherwise it is only stored through the `selectedAddress` state.
+   *
+   * @param {AddressFormValues} formValues address fields to update
+   */
+  const handleCreate = async ({ save, ...address }: AddressFormValues) => {
+    setError(undefined);
+    if (save) {
+      await addresses.methods.create(address, {
+        onSuccess: handleSelect,
+        onError: () => setError(intl.formatMessage(messages.actionCreation)),
+      });
+    } else {
+      handleSelect({
+        id: LOCAL_BILLING_ADDRESS_ID,
+        is_main: false,
+        ...address,
+      });
+    }
+  };
+
+  /**
+   * Update the `editedAddress` with new values provided as argument
+   * then clear `editedAddress` state if request succeeded.
+   *
+   * @param {AddressFormValues} formValues address fields to update
+   */
+  const handleUpdate = async ({ save, ...newAddress }: AddressFormValues) => {
+    setError(undefined);
+    addresses.methods.update(
+      {
+        ...editedAddress!,
+        ...newAddress,
+      },
+      {
+        onSuccess: () => setEditedAddress(undefined),
+        onError: () => setError(intl.formatMessage(messages.actionUpdate)),
+      },
+    );
+  };
+
+  /**
+   * Update the provided address to promote it as main
+   *
+   * @param {Joanie.Address} address
+   */
+  const promoteAddress = (address: Joanie.Address) => {
+    if (!address.is_main) {
+      setError(undefined);
+      addresses.methods.update(
+        {
+          ...address,
+          is_main: true,
+        },
+        {
+          onError: () => setError(intl.formatMessage(messages.actionPromotion)),
+        },
+      );
+    }
+  };
+
+  useEffect(() => {
+    setError(undefined);
+  }, [editedAddress]);
+
+  return (
+    <div className="AddressesManagement">
+      <button className="button button--tertiary" onClick={handleClose}>
+        <svg className="button__icon" role="img">
```

Comment:
You could add an `aria-hidden` attribute here to make sure no screen reader announce anything.